src/samples/audio/smart_amp_test_ipc4.c

```diff
@@ -346,42 +337,21 @@ static int smart_amp_prepare(struct processing_module *mod,
 	struct comp_buffer *sink_buffer;
 	struct list_item *blist;
 	int ret;
+	int i;
 
-	ret = smart_amp_params(mod);
```

Comment:
how is all the configuration now handled? Does this mean, that we're now only handling a single fixed configuration here? Are we losing any flexibility?

Reply:
@RanderWang can you put the #ifdef around this. Thanks.

Reply:
The param is set in module adapter layer, so remove it, please check https://github.com/thesofproject/sof/pull/8612. Why we need smart_amp_params(mod) to support different configurations ?  User can check "struct smart_amp_data" to get current config and process it.

And for source & sink buffer with loadable module, we can't set params in loadable module since no such interface is available.  BTW why users need to set such detail params ? They want a simple framework.

Reply:
do you mean, that after #8612 is merged, the configuration will be handled at the module-adapter level? Then this PR should be marked as dependent on that one, and maybe removing (then redundant) configuration setting here should be a separate commit